pkg/openshift/rosa/hcpvpc.go

```diff
@@ -0,0 +1,159 @@
+func copyFile(srcFile, destFile string) error {
+	srcReader, err := assets.FS.Open(srcFile)
+	if err != nil {
+		return fmt.Errorf("error opening %s file: %w", srcFile, err)
+	}
+	defer srcReader.Close()
+
+	destReader, err := os.Create(destFile)
+	if err != nil {
+		return fmt.Errorf("error creating runtime %s file: %w", destFile, err)
+	}
+	defer destReader.Close()
+
+	_, err = io.Copy(destReader, srcReader)
+	if err != nil {
+		return fmt.Errorf("error copying source file to destination file: %w", err)
+	}
+
+	return nil
+}
+
+// createHostedControlPlaneVPC creates the aws vpc used for provisioning hosted control plane clusters
+func (r *Provider) createHostedControlPlaneVPC(ctx context.Context, clusterName, awsRegion, workingDir string) (*vpc, error) {
+	action := "create"
+	var vpc vpc
+
+	if clusterName == "" || awsRegion == "" || workingDir == "" {
+		return nil, &hcpVPCError{action: action, err: errors.New("one or more parameters is empty")}
+	}
+
+	tf, err := terraform.New(ctx, workingDir)
+	if err != nil {
+		return nil, &hcpVPCError{action: action, err: fmt.Errorf("failed to construct terraform runner: %v", err)}
+	}
+
+	if err = tf.SetEnvVars(r.awsCredentials.CredentialsAsMap()); err != nil {
+		return nil, &hcpVPCError{action: action, err: fmt.Errorf("failed to set terraform runner aws credentials (env vars): %v", err)}
+	}
+
+	defer func() {
+		_ = tf.Uninstall(ctx)
+	}()
+
+	r.log.Info("Creating cluster", clusterNameLoggerKey, clusterName, "aws vpc in", awsRegionLoggerKey, awsRegion)
+
+	err = copyFile("terraform/setup-vpc.tf", fmt.Sprintf("%s/setup-vpc.tf", workingDir))
+	if err != nil {
+		return nil, &hcpVPCError{action: action, err: fmt.Errorf("failed to copy terraform file to working directory: %v", err)}
+	}
+
+	err = tf.Init(ctx)
+	if err != nil {
+		return nil, &hcpVPCError{action: action, err: fmt.Errorf("failed to perform terraform init: %v", err)}
+	}
+
+	err = tf.Plan(
+		ctx,
+		tfexec.Var(fmt.Sprintf("aws_region=%s", awsRegion)),
+		tfexec.Var(fmt.Sprintf("cluster_name=%s", clusterName)),
+	)
+	if err != nil {
+		return nil, &hcpVPCError{action: action, err: fmt.Errorf("failed to perform terraform plan: %v", err)}
+	}
+
+	err = tf.Apply(ctx)
+	if err != nil {
+		return nil, &hcpVPCError{action: action, err: fmt.Errorf("failed to perform terraform apply: %v", err)}
+	}
+
+	output, err := tf.Output(ctx)
+	if err != nil {
+		return nil, &hcpVPCError{action: action, err: fmt.Errorf("failed to perform terraform output: %v", err)}
+	}
+
+	vpc.privateSubnet = strings.ReplaceAll(string(output["cluster-private-subnet"].Value), "\"", "")
+	vpc.publicSubnet = strings.ReplaceAll(string(output["cluster-public-subnet"].Value), "\"", "")
+	vpc.nodePrivateSubnet = strings.ReplaceAll(string(output["node-private-subnet"].Value), "\"", "")
+
+	r.log.Info("Cluster", clusterNameLoggerKey, clusterName, "aws vpc created! Terraform state file resides at", terraformWorkingDirLoggerKey, workingDir)
```

Comment:
```suggestion
	r.log.Info("aws vpc created",  clusterNameLoggerKey, clusterName, terraformWorkingDirLoggerKey, workingDir)
```

bit nit-picky, a lot of these logs I think we can drop the `with` or words leading into the keys and just treat the statement as what happened then the key/values as extra context like:

```
- 	r.log.Info("Account roles deleted with", prefixLoggerKey, prefix)
+ 	r.log.Info("Account roles deleted", "accountRolesPrefix", prefix)
```

(or whatever the key is)

Reply:
Modified all the logging statements to be shorter and right to the point. Actual details about the message can be found in the key/value pairs logged.

Refer to 227aaace952ae7711b2c52f39d194176807d90cd